internal/daemon/controller/handlers/targets/target_service.go

```diff
@@ -1868,6 +1896,51 @@ func validateListRequest(ctx context.Context, req *pbs.ListTargetsRequest) error
 	return nil
 }
 
+func newOutputOpts(ctx context.Context, item target.Target, authResults auth.VerifyResults, authzScopes map[string]*scopes.ScopeInfo) []handlers.Option {
+	pr := perms.Resource{Id: item.GetPublicId(), ScopeId: item.GetProjectId(), Type: resource.Target}
+	outputFields := authResults.FetchOutputFields(pr, action.List).SelfOrDefaults(authResults.UserId)
+
+	outputOpts := make([]handlers.Option, 0, 3)
+	outputOpts = append(outputOpts, handlers.WithOutputFields(outputFields))
+
+	if outputFields.Has(globals.ScopeField) {
+		outputOpts = append(outputOpts, handlers.WithScope(authzScopes[item.GetProjectId()]))
+	}
+	if outputFields.Has(globals.AuthorizedActionsField) {
+		authorizedActions := authResults.FetchActionSetForId(ctx, item.GetPublicId(), IdActions, auth.WithResource(&pr)).Strings()
+		outputOpts = append(outputOpts, handlers.WithAuthorizedActions(authorizedActions))
+	}
+	return outputOpts
+}
+
+// parseRefreshToken parses a refresh token from the input, returning
+// an error if the parsing fails.
+func parseRefreshToken(ctx context.Context, token string) (*pbs.ListRefreshToken, error) {
+	const op = "list.parseRefreshToken"
+	marshaled, err := base58.Decode(token)
+	if err != nil {
+		return nil, errors.Wrap(ctx, err, op)
+	}
+	var tok pbs.ListRefreshToken
+	if err := proto.Unmarshal(marshaled, &tok); err != nil {
+		return nil, errors.Wrap(ctx, err, op)
+	}
+	return &tok, nil
+}
+
+// marshalRefreshToken marshals a refresh token to its string representation.
+func marshalRefreshToken(ctx context.Context, token *pbs.ListRefreshToken) (string, error) {
+	const op = "list.marshalRefreshToken"
+	if token == nil {
+		return "", errors.New(ctx, errors.InvalidParameter, op, "token is required")
+	}
+	marshaled, err := proto.Marshal(token)
+	if err != nil {
+		return "", errors.Wrap(ctx, err, op)
+	}
+	return base58.Encode(marshaled), nil
+}
```

Comment:
These functions will likely be moved to a separate package to unify logic across list handlers

Reply:
You could move these into the [handlers](https://github.com/hashicorp/boundary/tree/mikemountain-domain-rewrite/internal/daemon/controller/handlers) package. That is where we have some things used by all the handler implementations.